openapi-cli/src/main/java/io/ballerina/openapi/generators/service/RequestBodyGenerator.java

```diff
@@ -115,50 +114,72 @@ private TypeDescriptorNode getNodeForPayloadType(Components components, Map.Entr
+                            createToken(SyntaxKind.CLOSE_BRACKET_TOKEN));
+                    typeName = createArrayTypeDescriptorNode(createBuiltinSimpleNameReferenceNode(
+                                    null, createIdentifierToken(GeneratorConstants.BYTE)),
+                            NodeFactory.createNodeList(dimensionNode));
+                    break;
+                case APPLICATION_JSON:
+                    Schema<?> schema = components.getSchemas().get(schemaName);
+                    // This condition is avoid to wrong code generation for union type request body. If given schema
+                    // has oneOf type , then it will not be able to support via ballerina. We need to pick it mime
+                    // type instead of schema type.
+                    if ((schema instanceof ComposedSchema) && (((ComposedSchema) schema).getOneOf() != null)) {
                         identifierToken = createIdentifierToken(GeneratorConstants.JSON);
                         typeName = createSimpleNameReferenceNode(identifierToken);
-                }
-            } else {
-                typeName = getMediaTypeToken(mediaType);
+                    } else {
+                        typeName = createSimpleNameReferenceNode(createIdentifierToken(
+                                getValidName(schemaName, true)));
+                    }
+                    break;
+                case APPLICATION_URL_ENCODE:
+                    typeName = createSimpleNameReferenceNode(createIdentifierToken(
+                            getValidName(schemaName, true)));
+                    break;
+                default:
+                    identifierToken = createIdentifierToken(GeneratorConstants.JSON);
+                    typeName = createSimpleNameReferenceNode(identifierToken);
             }
         } else {
             typeName = getMediaTypeToken(mediaType);
         }
         return typeName;
     }
 
-    private SeparatedNodeList<MappingFieldNode> fillRequestAnnotationValues(Token comma, List<Node> literals,
-                                                                            IdentifierToken mediaType,
+    /**
+     *  This function fill the annotation values.
+     *
+     *  01. when field has some override media type
+     *  <pre> @http:Payload{mediaType: "application/x-www-form-urlencoded"} User payload</pre>
+     *  02. when field has list media value
+     *  <pre> @http:Payload{mediaType: ["application/json", "application/snowflake+json"]} User payload</pre>
+     *
+     */
+    private SeparatedNodeList<MappingFieldNode> fillRequestAnnotationValues(List<Node> literals,
                                                                             HashSet<Map.Entry<String,
                                                                                     MediaType>> equalDataType) {
+        Token comma = createToken(SyntaxKind.COMMA_TOKEN);
```

Comment:
Shall we remove this variable since its not being used?

Reply:
This is used inside the while loop, in line 188 🙂